openadapt/crud.py

```diff
@@ -109,28 +116,133 @@ def insert_recording(recording_data):
+    return db.query(Recording).filter_by(id=recording_id).first()
+
+
+def export_sql(recording_id):
+    """Export the recording data as SQL statements.
+
+    Args:
+        recording_id (int): The ID of the recording.
+
+    Returns:
+        str: The SQL statement to insert the recording into the output file.
+    """
+    engine = sa.create_engine(config.DB_URL)
+    Session = sessionmaker(bind=engine)
+    session = Session()
+
+    recording = get_recording_by_id(recording_id)
+
+    if recording:
+        sql = """
+            INSERT INTO recording
+            VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+        """
+        values = (
+            recording.id,
+            recording.timestamp,
+            recording.monitor_width,
+            recording.monitor_height,
+            recording.double_click_interval_seconds,
+            recording.double_click_distance_pixels,
+            recording.platform,
+            recording.task_description,
+        )
+
+        logger.info(f"Recording with ID {recording_id} exported successfully.")
+    else:
+        sql = ""
+        logger.info(f"No recording found with ID {recording_id}.")
+
+    return sql, values
+
+
+def create_db(recording_id, sql, values):
+    """Create a new database and import the recording data.
+
+    Args:
+        recording_id (int): The ID of the recording.
+        sql (str): The SQL statements to import the recording.
+
+    Returns:
+        tuple: A tuple containing the timestamp and the file path of the new database.
+    """
+    db.close()
+    db_fname = f"recording_{recording_id}.db"
+
+    timestamp = time.time()
+    source_file_path = config.ENV_FILE_PATH
+    target_file_path = f"{config.ENV_FILE_PATH}-{timestamp}"
+    logger.info(
+        f"source_file_path={source_file_path}, target_file_path={target_file_path}"
     )
+    shutil.copyfile(source_file_path, target_file_path)
+    config.set_db_url(db_fname)
+
+    with open(config.ENV_FILE_PATH, "r") as env_file:
+        env_file_lines = [
+            f"DB_FNAME={db_fname}\n"
+            if env_file_line.startswith("DB_FNAME")
+            else env_file_line
+            for env_file_line in env_file.readlines()
+        ]
+
+    with open(config.ENV_FILE_PATH, "w") as env_file:
+        env_file.writelines(env_file_lines)
+
+    engine = sa.create_engine(config.DB_URL)
+    Session = sessionmaker(bind=engine)
+    session = Session()
+    os.system("alembic upgrade head")
+    db.engine = engine
+
+    with engine.begin() as connection:
+        connection.execute(sql, values)
+
+    db_file_path = config.DB_FPATH.resolve()
+
+    return timestamp, db_file_path
+
+
+def restore_db(timestamp):
+    """Restore the database to a previous state.
+
+    Args:
+        timestamp (float): The timestamp associated with the backup file.
+    """
+    backup_file = f"{config.ENV_FILE_PATH}-{timestamp}"
+    shutil.copyfile(backup_file, config.ENV_FILE_PATH)
+    config.set_db_url("openadapt.db")
```

Comment:
Rather than setting it with a literal here, what do you think about saving the original value before overwriting it, then restoring it to the original value here?

Reply:
Addressed in https://github.com/MLDSAI/OpenAdapt/pull/223/commits/b801cafdaef9594eb46aab479ad87855a6cc3d4f